### PR TITLE
Feature/200 trading

### PR DESCRIPTION
**Pull request by sevap8**

### CODE REVIEW

**Comment by sevap8** [comment]
https://www.draw.io/?lightbox=1&highlight=0000ff&edit=_blank&layers=1&nav=1&title=Untitled%20Diagram.drawio#R7Vpbb6s4EP41PPYIDOTymFu3e7S7qtSs9vTRASfxHsCsMWlyfv3a2OYWkqZpSIhUqUrxYJvxeOb7ZjCGPQm3v1EYr%2F8kPgoMYPpbw54aAFgOAIb4M%2F2dlPQtVwpWFPuqUyF4wb%2BQEppKmmIfJZWOjJCA4bgq9EgUIY9VZJBS8lbttiRB9akxXKE9wYsHg33pP9hnaykdgH4hf0J4tdZPtnpDeSeEurNaSbKGPnkrieyZYU8oIUxehdsJCoTxtF3kuMcDd3PFKIrYKQNe3WQE6QQP8NP3x5%2Fu9DsLZg9qlg0MUrVgA%2FQCPt84FiqznbJD779U6DkOIV3hyLBH%2FK4Tb%2FmvWfrlHcT%2BoC17gAFeqX4eVxDRYg5%2BtVL%2Fsyct6oJUC%2BYURgn0GCbRE04YoTvdhS8zrQ%2Fjsr2pFrShV1yXrfd6vb%2F68oql%2FCFASyZvDsTN%2BnrjdBFgT%2BwI3zBg%2Fu6L8X0xfoVkPzPRF0Z%2FekzhK6j3wr0H0Y4rOUm5V4SdV%2FNlzaP0Lxiijmo5hQzNcaaevhzDXUeVlSYdhSSNWGbYjurpIw%2BHUCDwhCTsHCVBRT2wQZRhzkwjCa1TRmLeTQHtVKo1JrzXMsh4Zok5%2FtvjJeF2krxqAdV%2BhCEOBCM%2FoWCDxKwCBVkYiE75s8UD0fYg3Vg5iXH2RzwMmcBnUw1wbcV7mvh1%2B62gUeAo2bpMoVoIFXWv8rkLduMXiuA%2BQHbA3rMp8jnbqyahbE1WJILBrJCOKfczH4lphRsUff4gwvyZtf5FjO2UiWHKSNWWaIvZj9L1q5jqm6ta062aOWvsdCPi6%2F1RbpRGiWYxLGvpcXJ9YlHHd43bgKTUQ8eMpZIt7vSIHelnN3sBRQFkeFPV4%2FI72un05e8E0VnEMPtKW9pTT8%2BdxDDaUzlhFEcrwcApjT7Cv%2BXZbsYgufYdTh0KluNMBCOvq3rmtnxek%2BgsJe%2BajS1QZWPHamBjq4mNQVtsbH8CuxscIbN%2FgevHsbx8m3swoVDUmLKPoHsaYOEk70K%2BQPhR5GdpaHIM0DsD30LjUyG8BoIwFB4eLZK4urBOYGWp0LorhqpUM4x4P5MLctRdQ5ZTKyAcswGyzAbIGrZWPzi3rB90kXBK%2FXDBOsD5ZH6vhj4T6elqc3tObXPrRZ%2BsO9So2r7lapy%2FlU6nCwfJJ1%2Blw1ei3ZRoP1N8Xpp913TQt96nA2vYlMGabfGBuwcicw4Be3YWuFA1Bw9u%2FAsusg7CG2IBc5l27thwp2IXOA0kysj7m9K8czwf8LgbzzNe4Qh3EbMPNOntKhOUjN70Dq%2B1okGfHl6Xgi9Ip4NW6BTUy7u6z7dMp4ODdCrQoREWxY0H6eIC%2FKxeE%2FqJauUF0U0GeDmkyTkPoNqF62a3athBA%2Bg0lc1uawHQu%2FMAGLYSAIMaO9j1IqDlABi2FAAyE7xpCPRrqfrw5iHQv%2FMQsMxmg38yBuonbteOAe0EpSAY%2BT7PSKQTz8l8XcHzVr3Wsfvf3GrmAhocd%2BDuO%2B6gNccFt3Dc%2FP1B%2Fs7g1Si9Mnjn%2FLE4cnzV813n%2FNHSX3u1fQB5WjC51mnBNKI0%2B3JCd1Op%2FeHn6EPp2nMeP6hX4ZdSg8tG9t0cnN%2FCAQ8A0XVOwK2bvJO8uJkvXxGZtTi58gtG7T2fyAiB2ZQRlr6JvGla6ABQI1i3d%2BIL%2BnPeyPBm8ams3KTig2N79j8%3D

**Comment by sevap8** [comment]
I had some minor difficulties with the tests and I did not manage to insert some functionality into the launch method.
it works and carries out sales, data is added to the structure of transactions.

File: HW6/Seva_Popov/Trading/TradingSimulator.Core.Tests/TransactionTests.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace TradingSimulator.Core.Tests
+{
+    public class TransactionTests
```
Comment (chesdenis) [inline HW6/Seva_Popov/Trading/TradingSimulator.Core.Tests/TransactionTests.cs]: empty test class

Comment (sevap8) [reply HW6/Seva_Popov/Trading/TradingSimulator.Core.Tests/TransactionTests.cs]: Sorry to take so long. 
I added a test method AddTransactionHistory.
I had a question, do I understand correctly that we are not testing the transaction transaction?
There are random methods.

**Comment by sevap8** [comment]
I'm worried about this code:
`public class TransactionService
    {
        private readonly ITransactionRepositories transactionRepositories;
        Random random = new Random();

         public TransactionService(ITransactionRepositories transactionRepositories)
        {
            this.transactionRepositories = transactionRepositories;
        }

         public UserEntity SelectUser()
        {
            var users = transactionRepositories.GetUserList();
            int count = users.Count;
            return users[random.Next(count)];
        }


         public AddingSharesToThUserEntity SelectSharesToThUserEntity(UserEntity userEntity)
        {
            var usersAndShare = transactionRepositories.GetUserAndShareList(userEntity);
            int count = usersAndShare.Count; 
            return usersAndShare[random.Next(count)]; 
        }

         public SharesEntity FindOutTheValueOfShares(AddingSharesToThUserEntity addingSharesToThUserEntity)
        {
            var shares = transactionRepositories.GetShares(addingSharesToThUserEntity);
            return shares[0];
        }

         public AddingSharesToThUserEntity SelectStocksForUserObjectParameters(UserEntity userEntity, SharesEntity sharesEntity)
        {
            var usersAndShare = transactionRepositories.GetUserAndShareParameters(userEntity, sharesEntity);
            int count = usersAndShare.Count;
            try
            {
                return usersAndShare[0];
            }
            catch (System.ArgumentOutOfRangeException)
            {
                AddingSharesToThUserEntity addingSharesToThUserEntity = new AddingSharesToThUserEntity();
                return addingSharesToThUserEntity;
            }

         }

         public int RandomNumberGenerator (int number)
        {
           return random.Next(number);
        }`

File: HW6/Seva_Popov/Trading/TradingSimulator.Core/Services/TransactionService.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using TradingSimulator.Core.Models;
+using TradingSimulator.Core.Repositories;
+
+namespace TradingSimulator.Core.Services
+{
+    public class TransactionService
```
Comment (sevap8) [inline HW6/Seva_Popov/Trading/TradingSimulator.Core/Services/TransactionService.cs]: I'm worried about this code

**Comment by chesdenis** [comment]
> I'm worried about this code:
> `public class TransactionService
> {
> private readonly ITransactionRepositories transactionRepositories;
> Random random = new Random();
> 
> ```
>      public TransactionService(ITransactionRepositories transactionRepositories)
>     {
>         this.transactionRepositories = transactionRepositories;
>     }
> 
>      public UserEntity SelectUser()
>     {
>         var users = transactionRepositories.GetUserList();
>         int count = users.Count;
>         return users[random.Next(count)];
>     }
> 
> 
>      public AddingSharesToThUserEntity SelectSharesToThUserEntity(UserEntity userEntity)
>     {
>         var usersAndShare = transactionRepositories.GetUserAndShareList(userEntity);
>         int count = usersAndShare.Count; 
>         return usersAndShare[random.Next(count)]; 
>     }
> 
>      public SharesEntity FindOutTheValueOfShares(AddingSharesToThUserEntity addingSharesToThUserEntity)
>     {
>         var shares = transactionRepositories.GetShares(addingSharesToThUserEntity);
>         return shares[0];
>     }
> 
>      public AddingSharesToThUserEntity SelectStocksForUserObjectParameters(UserEntity userEntity, SharesEntity sharesEntity)
>     {
>         var usersAndShare = transactionRepositories.GetUserAndShareParameters(userEntity, sharesEntity);
>         int count = usersAndShare.Count;
>         try
>         {
>             return usersAndShare[0];
>         }
>         catch (System.ArgumentOutOfRangeException)
>         {
>             AddingSharesToThUserEntity addingSharesToThUserEntity = new AddingSharesToThUserEntity();
>             return addingSharesToThUserEntity;
>         }
> 
>      }
> 
>      public int RandomNumberGenerator (int number)
>     {
>        return random.Next(number);
>     }`
> ```

yes, this random it is not a part of core logic,. you should move it into simulator.

**Comment by sevap8** [comment]
Added by Trading Simulation.

**Pull request merged**